src/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -268,6 +268,9 @@ public SamReader open(final SamInputResource resource) {
                             bufferedStream.close();
                             bufferedStream = null;
                         }
+
+                        bufferedStream = IOUtil.maybeBufferInputStream(indexMaybe.asUnbufferedInputStream());
```

Comment:
Actually, instead of opening an input stream on the index here, recommend switching to the constructor:

        public CRAMFileReader(final File cramFile, final File indexFile, final ReferenceSource referenceSource)

as I'm not sure I trust the constructors that take a stream, since the javadoc for those constructors contradicts itself (in one place the `inputStream` is described as `CRAM stream to read` and in another place that same parameter is described as `index file to be used for random access`).